src/code/sys_math3d.c

```diff
@@ -0,0 +1,1691 @@
+#include <ultra64.h>
+#include <global.h>
+#include <vt.h>
+#include <sys_math3d.h>
+
+s32 func_800CA7D0(f32 arg0, f32 arg1, f32 arg2, f32 arg3, f32 arg4, f32 arg5, f32 arg6, f32 arg7, Vec3f* arg8,
+                  Vec3f* arg9, Vec3f* argA) {
+    Vec3f sp34;
+    static Linef D_8016A5A0;
+    static Linef D_8016A5B8;
+    if (func_800CAD08(arg0, arg1, arg2, arg3, arg4, arg5, arg6, arg7, &D_8016A5A0) == 0) {
+        return 0;
+    }
+    Math_Vec3f_Copy(&D_8016A5B8.a, &D_8016A5A0.a);
+
+    VEC3F(D_8016A5B8.b, ((D_8016A5A0.b.x * 100.0f) + D_8016A5A0.a.x), ((D_8016A5A0.b.y * 100.0f) + D_8016A5A0.a.y),
+          ((D_8016A5A0.b.z * 100.0f) + D_8016A5A0.a.z));
+
+    if (!func_800CA8E8(&D_8016A5B8.a, &D_8016A5B8.b, arg8, arg9, argA, &sp34)) {
+        return 0;
+    }
+    return 1;
+}
+
+#pragma GLOBAL_ASM("asm/non_matchings/code/sys_math3d/func_800CA8E8.s")
+
+void Math3D_LineVsPos(Linef* line, Vec3f* pos, Vec3f* ret) {
+    f32 temp_ret;
+    f32 temp_f0;
+
+    temp_ret = func_800CB600(&line->b);
+    if (fabsf(temp_ret) < 0.008f) {
+        osSyncPrintf(VT_COL(YELLOW, BLACK));
+        osSyncPrintf("Math3D_lineVsPosSuisenCross():直線の長さがありません\n"); // Math3D_lineVsPosSuisenCross(): No
```

Comment:
in the rest of the codebase we tend to put translation comments above the osSyncPrintf, think it looks nicer and eliminates the need for empty newlines